apps/src/lib/levelbuilder/lesson-editor/ActivitySectionCardButtons.jsx

```diff
@@ -0,0 +1,154 @@
+    addTip: PropTypes.func,
+    editTip: PropTypes.func,
+    addLevel: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      addTipOpen: false,
+      editingExistingTip: false,
+      addResourceOpen: false,
+      addLevelOpen: false,
+      tipToEdit: {key: 'tip', type: 'teachingTip', markdown: ''}
+    };
+  }
+
+  handleOpenAddLevel = () => {
+    this.setState({addLevelOpen: true});
+  };
+
+  handleCloseAddLevel = () => {
+    this.setState({addLevelOpen: false});
+  };
+
+  handleEditTip = tip => {
+    this.setState({tipToEdit: tip, addTipOpen: true, editingExistingTip: true});
+  };
+
+  handleOpenAddTip = () => {
+    this.setState({
+      tipToEdit: {
+        key: this.generateTipKey(),
+        type: 'teachingTip',
+        markdown: ''
+      },
+      addTipOpen: true
+    });
+  };
+
+  generateTipKey = () => {
+    let tipNumber = this.props.activitySection.tips.length + 1;
+    while (
+      this.props.activitySection.tips.some(
+        tip => tip.key === `tip-${tipNumber}`
+      )
+    ) {
+      tipNumber++;
+    }
+
+    return `tip-${tipNumber}`;
+  };
+
+  handleCloseAddTip = tip => {
+    if (!this.state.editingExistingTip) {
+      this.props.addTip(tip);
+    }
+    this.setState({addTipOpen: false, editingExistingTip: false});
+  };
+
+  handleOpenAddResource = () => {
+    this.setState({addResourceOpen: true});
+  };
+
+  handleCloseAddResource = () => {
+    this.setState({addResourceOpen: false});
+  };
+
+  render() {
+    return (
+      <div>
+        <div style={styles.bottomControls}>
+          <span>
+            <button
+              onMouseDown={this.handleOpenAddLevel}
+              className="btn"
+              style={styles.addLevel}
+              type="button"
+            >
+              <i style={{marginRight: 7}} className="fa fa-plus-circle" />
+              Add Level
+            </button>
+            <button
+              onMouseDown={this.handleOpenAddTip}
+              className="btn"
+              style={styles.addLevel}
+              type="button"
+            >
+              <i style={{marginRight: 7}} className="fa fa-plus-circle" />
+              Add Tip
+            </button>
+            <button
+              onMouseDown={this.handleOpenAddResource}
+              className="btn"
+              style={styles.addLevel}
+              type="button"
+            >
+              <i style={{marginRight: 7}} className="fa fa-plus-circle" />
+              Add Resource Link
+            </button>
+          </span>
+          {this.props.activitySection.tips.length > 0 && (
+            <span>
+              {this.props.activitySection.tips.map(tip => {
+                return (
+                  <LessonTipIconWithTooltip
+                    tip={tip}
+                    key={tip.key}
+                    onClick={this.handleEditTip}
+                  />
+                );
+              })}
```

Comment:
nit: this can be written more compactly as:
```js
              {this.props.activitySection.tips.map(tip => (
                <LessonTipIconWithTooltip
                  ...
                />
              ))}